### ISSUE TITLE
Pygame sometimes fails to set icon

### ISSUE THREAD

**Issue by lordmauve** [lordmauve/pgzero#65]
Originally reported by: **Daniel Pope (Bitbucket: [lordmauve](https://bitbucket.org/lordmauve), GitHub: [lordmauve](https://github.com/lordmauve))**

----------------------------------------

Pygame Zero supports setting the icon by assigning to the `ICON` constant.

Unfortunately, this is only picked up after a window has been created. From the [Pygame documentation](https://www.pygame.org/docs/ref/display.html#pygame.display.set_icon):

> Some systems do not allow the window icon to change after it has been shown. This function can be called before pygame.display.set_mode() to create the icon before the display mode is set.

We must create a window prior to executing the module in order to support `Surface.convert_alpha()` which is used by the image loader. We might be able to statically analyse the module in order to identify values for WIDTH, HEIGHT and ICON and avoid having to change them.

----------------------------------------
- Bitbucket: https://bitbucket.org/lordmauve/pgzero/issue/65

**Comment by lucas-james** [comment]
I'll take a look into this 👍

**Comment by lucas-james** [comment]
@lordmauve can this issue be closed?

**Comment by lordmauve** [comment]
I don't believe this issue is fixed. We definitely get an icon now but it isn't always the icon defined in the script. It's relatively hard to fix this; one way would be to statically analyse the source code looking for an icon definition like

```python
ICON = 'name'
```

and this would work most of the time, but would have surprising consequences - for example that the above is not equivalent to

```python
ICON = f'name'
```

**Comment by mjovanc** [comment]
I can concur that this issue do exist for me as well. I'm using pygame==1.9.6 with Python 3.7.5. I tried to set my icon with this code:

```
windowIcon = pygame.image.load(os.path.join(RESOURCES_DIR, 'img/icon.png'))
pygame.display.set_icon(windowIcon)
```

**Comment by MilanLizriadi** [comment]
ICON = "icon_name"

**Comment by MilanLizriadi** [comment]
`ICON = "stand"`

**Comment by MilanLizriadi** [comment]
```
ICON = "stand"
```

**Comment by MilanLizriadi** [comment]
```python
ICON = "stand"
```

**Comment by milanlizriadi** [comment]
```python
ICON = "images/icon.png"
```
